starboard/BUILD.gn

```diff
@@ -24,6 +24,7 @@ group("gn_all") {
     "//starboard/client_porting/eztime:eztime_test",
     "//starboard/client_porting/icu_init",
     "//starboard/client_porting/poem:poem_unittests",
+    "//starboard/contrib/cast/cast_starboard_api/samples:cast",
```

Comment:
I don't know if we want to unconditionally include anything in `/contrib/` by default in the `all` build profile. I think this should perhaps be gated on a GN arg e.g. `contrib_build_with_cast` or such ? @y4vor @andrewsavage1 thoughts ?

Reply:
Depends what cast wants to build on—if they only care about linux modular builds, that's fine, but if they care about many other platforms not building by default could cause our build matrix to blow up

Reply:
We are using the modular build by design as libcast_starboard_api.so will always be linked dynamically and is a thin wrapper on top of libstarboard_platform_group.so.

I'll make it so this is only included in the modular build.